hoomd/state.py

```diff
@@ -158,20 +160,23 @@ def special_pair_types(self):
     @property
     def box(self):
         b = self._cpp_sys_def.getParticleData().getGlobalBox()
-        L = b.getL()
-        return boxdim(Lx=L.x, Ly=L.y, Lz=L.z,
-                      xy=b.getTiltFactorXY(),
-                      xz=b.getTiltFactorXZ(),
-                      yz=b.getTiltFactorYZ(),
-                      dimensions=self._cpp_sys_def.getNDimensions())
-
-    # Set the system box
-    # \param value The new boundaries (a data.boxdim object)
+        return Box._from_cpp(b)
+
     @box.setter
     def box(self, value):
-        if not isinstance(value, boxdim):
-            raise TypeError('box must be a data.boxdim object')
-        self._cpp_sys_def.getParticleData().setGlobalBox(value._getBoxDim())
+        """Set the system box.
+
+        Args:
+            value (hoomd.box.Box): The new box.
+        """
+        if not isinstance(value, Box):
+            raise TypeError('box must be a hoomd.box.Box object')
+        if value.dimensions != self._cpp_sys_def.getNDimensions():
+            warn("Box changing dimensions from {} to {}."
+                 "".format(self._cpp_sys_def.getNDimensions(),
+                           value.dimensions), UserWarning)
```

Comment:
I'm assuming that `warnings.warn` will write output to stderr which will repeat (and potentially mix) output from all MPI ranks. Use the Messenger.warning functionality to issue the warning only from the root rank.

Reply:
In any case, `UserWarning` results in import errors with the current builds:
```
hoomd/state.py:3: in <module>
    from warnings import warn, UserWarning
E   ImportError: cannot import name 'UserWarning' from 'warnings' (/usr/lib/python3.7/warnings.py)
```